cypress/utils/mock-web3-provider.ts

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { decrypt, personalSign } from '@metamask/eth-sig-util'
+
+type ProviderSetup = {
+  address: string
+  privateKey: string
+  networkVersion: number
+  debug?: boolean
+  manualConfirmEnable?: boolean
+}
+
+type RequestMethod =
+  | 'eth_requestAccounts'
+  | 'eth_accounts'
+  | 'net_version'
+  | 'eth_chainId'
+  | 'personal_sign'
+  | 'eth_decrypt'
+  | 'eth_blockNumber'
+  | 'eth_sendTransaction'
+  | 'eth_getTransactionByHash'
+
+interface IMockProvider {
+  request(args: { method: 'eth_accounts'; params: string[] }): Promise<string[]>
+  request(args: {
+    method: 'eth_requestAccounts'
+    params: string[]
+  }): Promise<string[]>
+
+  request(args: { method: 'net_version' }): Promise<number>
+  request(args: { method: 'eth_chainId'; params: string[] }): Promise<string>
+
+  request(args: { method: 'personal_sign'; params: string[] }): Promise<string>
+  request(args: { method: 'eth_decrypt'; params: string[] }): Promise<string>
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request(args: { method: string; params?: any[] }): Promise<any>
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export class MockProvider implements IMockProvider {
```

Comment:
based effort here!!!

Reply:
cant take the credit for most of this - found it online, but had to enhance it for our purposes